sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/quickpulse/QuickPulseDataCollector.java

```diff
@@ -140,16 +191,22 @@ private void addDependency(RemoteDependencyData telemetry, int itemCount) {
         if (success != null && !success) { // success should not be null
             counters.unsuccessfulRdds.incrementAndGet();
         }
-        RemoteDependency dependencyDoc = new RemoteDependency();
-        dependencyDoc.setName(telemetry.getName());
-        dependencyDoc.setCommandName(telemetry.getData());
-        dependencyDoc.setDuration(Duration.ofMillis(durationMillis).toString());
-        dependencyDoc.setResultCode(telemetry.getResultCode());
-        dependencyDoc.setProperties(setCustomDimensions(telemetry.getProperties(), telemetry.getMeasurements()));
-
-        synchronized (counters.documentList) {
-            if (counters.documentList.size() < Counters.MAX_DOCUMENTS_SIZE) {
-                counters.documentList.add(dependencyDoc);
+
+        DependencyDataColumns columns = new DependencyDataColumns(telemetry);
+        applyMetricFilters(columns, "Dependency");
+
+        if (matchesDocumentFilters(columns, "Dependency")) {
```

Comment:
I'd call `configuration.get()` once and pass the same instance to both `applyMetricFilters` and `matchesDocumentFilters` so that you can't end up handling one request using two different configurations (in case a new configuration comes in async between these calls)